### DISCUSSION TITLE
Export only selected columns

### DISCUSSION THREAD

**Discussion by aeq-dev** [rappasoft/laravel-livewire-tables#853] [Q&A]
Hello,
Is there a way to get selected columns in order to only export them ? Thanks

**Comment by nafil161** [comment]
I am also searching for this.

> Hello, Is there a way to get selected columns in order to only export them ? Thanks

I am also searching for this. or anyway we can get the column status. then from there we can do some code to add in export select
?

**Comment by dgillier** [comment]
There is a setColumnSelectStatus but not a getColumnSelectStatus.

But you get the status by looking @ the class data ($this->columns), the columns are in a collection with a "selected" field:

```
#columns: Illuminate\Support\Collection {#1865 ▼
    #items: array:14 [▼
      0 => Rappasoft\LaravelLivewireTables\Views\Column {#1654 ▼
        #component: App\Http\Livewire\Expenditure\ExpenditureTable {[#1730](http://localhost/members/expenditures#sf-dump-477579639-ref21730)}
        #title: "Id"
        #hash: "b80bb7740288fda1f201890375a60c8f"
        #from: "id"
        #field: "id"
        #table: "expenditures"
        #relations: []
        #sortable: true
        #sortCallback: null
        #searchable: false
        #searchCallback: null
        #collapseOnMobile: true
        #collapseOnTablet: false
        #sortingPillTitle: null
        #sortingPillDirectionAsc: null
        #sortingPillDirectionDesc: null
        #eagerLoadRelations: false
        #formatCallback: Closure($value, $row, $column) {#1787 ▶}
        #html: false
        #labelCallback: null
        #hidden: false
        #selectable: true
        #selected: false
        #secondaryHeader: false
        #secondaryHeaderCallback: null
        #footer: false
        #footerCallback: null
        #clickable: true
      }
```

Hope this help. Denis

**Reply by aeq-dev** [reply]
Thanks @dgillier for reply, unfortunately it doesn't work, the field selected = true even not selected

**Comment by dgillier** [comment]
@aeq-dev Strange, it's working fine for me (as my copied data show an unselected column as false) , which version are you using ?
